pages/api/account.ts

```diff
@@ -31,16 +31,23 @@ export default async (req: NextApiRequest, res: NextApiResponse) => {
       throw "Incorrect hash algorithm"
     }
 
+    const apiKey = req.body["api-key"]
```

Comment:
```suggestion
    const apiKey = req.headers["Authorization"]
```

I think it'd be better to pass the API key in through the `Authorization` header rather than the request body.

You can still conditionally require the `h-captcha-response` field depending on whether or not the authorization header is set [by using the `$` context format](https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema):

```js
const captchaSchemaShape = {
  "h-captcha-response": yup.string().when('$apiKey', {
     is: (val: string) => !val || val.length === 0,
     then: yup.string().required(),
     otherwise: yup.string(),
  })
}

// ...

const apiKey = req.headers["Authorization"]
await createAccountSchemaServer.validate(req.body, { context: { apiKey } })